connectors/filtering/basic_rule.py

```diff
@@ -94,6 +94,52 @@ def try_coerce(value):
     return coerced_value
 
 
+class RuleMatchStats:
+    def __init__(self, policy, matched_docs_count):
```

Comment:
```suggestion
    def __init__(self, policy, matched_docs_count=0):
```

Probably could have a default init count